Backend: code improvements / cleanups

rueblimaster opened the pull request:
## What
I just fixed some pretty tame intelliJ warnings for the most part.
All changes here shouldn't change a thing in the logic.

## Changelog Technical Details
+ Cleaned up some code. - rueblimaster

rueblimaster committed:
Update GardenPlotIcon.kt

rueblimaster committed:
Update Renderable.kt

rueblimaster committed:
Update ShiftClickNpcSell.kt

rueblimaster committed:
reduce to wordy comments

rueblimaster committed:
remove + in string

rueblimaster committed:
minor warning fixes

rueblimaster committed:
Merge branch 'hannibal002:beta' into minor-code-improvements

rueblimaster committed:
Merge remote-tracking branch 'upstream/beta' into minor-code-improvements

rueblimaster commented:
also pr #3600 !!!

Bloxigus commented on src/main/java/at/hannibal2/skyhanni/utils/EntityOutlineRenderer.kt:
```diff
@@ -203,7 +203,7 @@ object EntityOutlineRenderer {
         GlStateManager.enableAlpha()
 
-        return !shouldRenderOutlines
+        return shouldRenderOutlines
```
???

DavidArthurCole commented on src/main/java/at/hannibal2/skyhanni/utils/CollectionUtils.kt:
```diff
@@ -361,9 +361,9 @@ object CollectionUtils {
-        if (emptySpace < 0) {
-            //    throw IllegalArgumentException("Not enough space for content")
-        }
+//         if (emptySpace < 0) {
```
??? part 2

rueblimaster replied:
the if statement does nothing, so now I commented it out, I can delete also just delete it, if that's preferred?

DavidArthurCole commented on src/main/java/at/hannibal2/skyhanni/utils/CollectionUtils.kt:
```diff
@@ -372,9 +372,9 @@ object CollectionUtils {
-        if (emptySpace < 0) {
-            //    throw IllegalArgumentException("Not enough space for content")
-        }
+//         if (emptySpace < 0) {
```
??? part 3

DavidArthurCole commented on src/main/java/at/hannibal2/skyhanni/utils/EssenceUtils.kt:
```diff
@@ -143,12 +143,12 @@ object EssenceUtils {
 
     private fun extractPurchasedUpgrades(inventoryItems: Map<Int, ItemStack>) = buildMap {
-        for (value in inventoryItems.values) {
+        loop@ for (value in inventoryItems.values) {
```
Why? This changes nothing

rueblimaster replied:
to remove the ambiguity of the continue statements

DavidArthurCole commented on src/main/java/at/hannibal2/skyhanni/utils/FakePlayer.kt:
```diff
@@ -5,10 +5,11 @@ import net.minecraft.client.entity.EntityOtherPlayerMP
 class FakePlayer : EntityOtherPlayerMP(MinecraftCompat.localWorld, MinecraftCompat.localPlayer.gameProfile) {
 
-    override fun getLocationSkin() =
+    override fun getLocationSkin(): ResourceLocation =
```
again, changes nothing, and signature typing is not necessary on overrides

rueblimaster replied:
"Declaration has type inferred from a platform call, which can lead to unchecked nullability issues. Specify type explicitly as nullable or non-nullable."

hannibal002 replied:
This is a valid point imo

SuperCl4sh commented on src/main/java/at/hannibal2/skyhanni/utils/EntityOutlineRenderer.kt:
```diff
@@ -219,7 +219,7 @@ object EntityOutlineRenderer {
     /**
      * Caches OptiFine settings and determines whether outlines should be rendered
      *
-     * @return `true` iff outlines should be rendered
```
I'm not sure this is a typo. iff is short for if and only if.

rueblimaster replied:
good point
tho I think in this case, the meaning of the sentence doesn't change at all, right?

SuperCl4sh replied:
If you mean if is the same as iff, then technically no. If A then B means that when A is satisfied, B is satisfied. A iff B means that if A is satisfied, then B is satisfied AND if B is satisfied, then A is satisfied. The existing comment is fine (and technically more correct).

rueblimaster replied:
ye, sure

DavidArthurCole commented on src/main/java/at/hannibal2/skyhanni/utils/renderables/Renderable.kt:
```diff
@@ -725,11 +725,10 @@ interface Renderable {
-                    val yOff: Int
-                    if (shouldRenderTopElseBottom) {
-                        yOff = 0
+                    val yOff: Int = if (shouldRenderTopElseBottom) {
```
```suggestion
                    val yOff: Int = 
                    	if (shouldRenderTopElseBottom) 0
                    	else content.height + ySpacing
```

rueblimaster committed:
even stricter detekt part 1

Bloxigus commented:
I mean no offense, but how much chatgpt/copilot/etc did you use to make this pr

rueblimaster committed:
delete more commented out code

rueblimaster committed:
more readable code design

Co-authored-by: David Cole <40234707+DavidArthurCole@users.noreply.github.com>

rueblimaster committed:
Revert "more readable code design"

This reverts commit d0d26d7bbc45af77d0481037c84e35df67dddb32.

rueblimaster committed:
again. but correct this time

rueblimaster commented:
> I mean no offense, but how much chatgpt/copilot/etc did you use to make this pr

none of the actually, but I did use the IntelliJ Quick-Fixes

rueblimaster committed:
Update EssenceUtils.kt

hannibal002 commented on src/main/java/at/hannibal2/skyhanni/features/garden/inventory/plots/GardenPlotIcon.kt:
```diff
@@ -101,9 +101,7 @@ object GardenPlotIcon {
-            return
-        }
-        if (editMode != 0) {
+        } else if (editMode != 0) {
```
insetad use a early return here (`if (editMode == 0) return`)

hannibal002 commented on src/main/java/at/hannibal2/skyhanni/utils/renderables/Renderable.kt:
```diff
@@ -725,12 +725,9 @@ interface Renderable {
-                    } else {
-                        yOff = content.height + ySpacing
-                    }
+                    val yOff: Int =
```
remove `: Int` and the line breaks

hannibal002 reviewed (changes requested):
Interesting random fixes. some of those stuff are still not perfect.

rueblimaster committed:
strictest detekt

rueblimaster committed:
add middle mouse

rueblimaster committed:
add PrimitiveItemStack.toPrimitiveIngredient

CalMWolfs commented on src/main/java/at/hannibal2/skyhanni/features/inventory/ShiftClickNpcSell.kt:
```diff
@@ -17,7 +17,7 @@ object ShiftClickNpcSell {
     private val config get() = SkyHanniMod.feature.inventory.shiftClickNPCSell
 
-    private val sellSlot = -4
+    private const val sellSlot = -4
```
```suggestion
    private const val SELL_SLOT = -4
```
make sure to rename it where it is used as well

rueblimaster committed:
correct const naming

rueblimaster committed:
Merge branch 'beta' into minor-code-improvements

rueblimaster committed:
Merge remote-tracking branch 'upstream/beta' into minor-code-improvements

rueblimaster committed:
Merge branch 'beta' into minor-code-improvements

rueblimaster committed:
Merge branch 'beta' into minor-code-improvements

rueblimaster committed:
Merge branch 'beta' into minor-code-improvements

rueblimaster committed:
Merge branch 'beta' into minor-code-improvements

rueblimaster committed:
Merge branch 'beta' into minor-code-improvements

rueblimaster committed:
Merge branch 'beta' into minor-code-improvements

rueblimaster committed:
Merge branch 'beta' into minor-code-improvements

rueblimaster committed:
Merge branch 'beta' into minor-code-improvements

rueblimaster committed:
Merge branch 'beta' into minor-code-improvements

rueblimaster committed:
Merge branch 'beta' into minor-code-improvements

rueblimaster committed:
Merge remote-tracking branch 'upstream/beta' into minor-code-improvements

rueblimaster committed:
Refactor GardenPlotIcon event handling logic

Simplifies and restructures the event handling logic in GardenPlotIcon.kt by consolidating edit mode checks and improving control flow for slot interactions. This makes the code easier to follow and maintain.

rueblimaster committed:
Merge branch 'beta' into minor-code-improvements

rueblimaster committed:
Merge branch 'beta' into minor-code-improvements

rueblimaster committed:
Merge branch 'beta' into minor-code-improvements

hannibal002 commented on src/main/java/at/hannibal2/skyhanni/features/garden/inventory/plots/GardenPlotIcon.kt:
```diff
@@ -101,30 +101,26 @@ object GardenPlotIcon {
+                plotList.remove(event.slotId)
                 return
             }
-            if (event.slotId != 53) {
```
why dont we check for id 53 anymore?

rueblimaster replied:
because it already gets checked in line 93 or so

hannibal002 committed:
Merge branch 'beta' into fork/rueblimaster/minor-code-improvements

# Conflicts:
#	src/main/java/at/hannibal2/skyhanni/utils/FakePlayer.kt

rueblimaster committed:
Merge branch 'beta' into minor-code-improvements

hannibal002 committed:
Merge branch 'beta' into fork/rueblimaster/minor-code-improvements

hannibal002 reviewed (approved):
lgtm
